### ISSUE TITLE
Search bar in NTP - Visual adjustments

### ISSUE THREAD

**Issue by aguscruiz** [brave/brave-browser#38704]
### Description

Follow up to this issue
- https://github.com/brave/brave-browser/issues/35535
- https://github.com/brave/brave-browser/issues/38506

I have the library file here for specifics
https://www.figma.com/design/QCBklDojDtDTKApz10XqSQ/%F0%9F%97%82-New-Tab-Page?node-id=1-608&t=4z8KmmCUWNq5OuDj-1

### Steps to reproduce

Open NTP and see search widget

### Collapsed
<img width="1310" alt="image" src="https://github.com/brave/brave-browser/assets/15734630/8b66d339-7807-490f-9891-52ba7c43846b">

### Expanded
<img width="1317" alt="image" src="https://github.com/brave/brave-browser/assets/15734630/ca4b097c-dbf3-43ea-8c5a-d124eb72e095">

### Expected result

https://www.figma.com/design/lVTufBkl5eNrtOQjlAHxjb/Search-box-in-NTP?node-id=1-2

### Collapsed
![image](https://github.com/brave/brave-browser/assets/15734630/11b58b62-0e1f-4ee7-a382-017993dda6d1)

- [x] Dropdown carat for engine needs to be glass/50%
- [x] Same for the magnifying glass icon on the right
- [x] The border and background around the dropdown button should be removed
- [ ] Placeholder text needs to be glass/70%
- [ ] The border around the search box looks kinda dark gray, should have that gradient outline like in Figma
- [x] The border radius was changed from 8px to 16px at search.brave.com so we should match.
- [ ] The height of the search box was changed to 58px at some point too.
- [ ] Removed the dropdown arrow and divider on the right. To open the dropdown you now have to click on the search engine icon.
- [ ] Padding around items had to be adjusted to match the alignment of the autosuggest results.
- [ ] The height of the search box was changed to 58px.
- [ ] Change the contextual menu icon (3 dots) to the vertical version

### Expanded
![image](https://github.com/brave/brave-browser/assets/15734630/e6f8e1ca-cf23-4853-9970-515d160d9c97)

- [ ] Center verticall like in proposed design https://www.figma.com/design/lVTufBkl5eNrtOQjlAHxjb/Search-box-in-NTP?node-id=971-13887&t=6WEeTbqKn1fRtLZI-1
- [x] When the magnifying glass icon appears, it shouldn't have a background and the icon color is glass/50%
- [x] The "Ask Leo" action should have Leo's icon and background
- [x] There should be a divider above "Ask Leo"
- [x] Hover effect on items should be Glass/10%
- [ ] The border radius was changed from 8px to 16px at search.brave.com so we should match.
- [ ] Padding between items and container is now 8px. Spacing between items is 4px.
- [ ] Padding within each item is now 8px horizontally too. Vertically is 4px
- [ ] Add the close (X) button and divider when a search term is entered

👇
Basically everything needs to vertically align like this
![image](https://github.com/brave/brave-browser/assets/15734630/9b4109d9-f711-46f9-8da2-fd2c2812df7e)

### Search engine selector
![image](https://github.com/brave/brave-browser/assets/15734630/d98f9706-4c39-4c11-9088-104efdedb416)

- [ ] Bug: Clicking on the search engine icon opens the full view, should let me choose the engine
- [ ] Bing icon should change to the new `microsoft-color` icon
- [ ] Dropdown should be 180px wide.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

1.68.55 Chromium: 126.0.6478.17 (Official Build) nightly (arm64) 

### Channel information

- [ ] release (stable)
- [ ] beta
- [X] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by fallaciousreasoning** [comment]
@aguscruiz could you check the latest Nightly - it looks quite different to the screenshots you posted

**Comment by aguscruiz** [comment]
Oh great, I just got the update. Looks way better! I'll check the items that are fixed in this latest version. 

<img width="634" alt="image" src="https://github.com/brave/brave-browser/assets/15734630/741eb137-7deb-490a-82ec-5eee6ce3fa63">

<img width="606" alt="image" src="https://github.com/brave/brave-browser/assets/15734630/d02abd46-5ae9-41de-bfea-3b04e82c22a9">

-----------------------
Ok done, still a couple to do, to be fair most are from the changes asked for the engines dropdown.

**Comment by aguscruiz** [comment]
@fallaciousreasoning I had to add two more items 

- The border radius was changed from 8px to 16px at search.brave.com so we should match.
- The height of the search box was changed to 58px at some point too.

Seems like this was changed in search.brave.com after I started working on this, so I had to update ours. 

I won't change anything else until we have this one in the bag. Then we can do some improvements that I got as feedback, is that ok?
